x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/UpdateInferenceModelAction.java

```diff
@@ -70,8 +70,8 @@ public Request(String inferenceEntityId, BytesReference content, XContentType co
         public Request(StreamInput in) throws IOException {
             super(in);
             this.inferenceEntityId = in.readString();
-            this.content = in.readBytesReference();
             this.taskType = TaskType.fromStream(in);
+            this.content = in.readBytesReference();
```

Comment:
Surprisingly the fix doesn't require a new transport version. This change means that reading from older nodes will start working and reads from same version and future versions will work

Reply:
Making sure I understand, we don't need a new transport version because it was incorrect to start? Will this have any expected impact on serverless?

Reply:
The writeTo() code writes `taskType` then `content` and the read was reading `content` then `taskType`.
My intuition was that a fix would need a new transport version but when I wrote the code it ended up as:
```
if (version.before(MY_FIX_VERSION)) {
  // change the order as this is what write is doing in the old version
  this.taskType = TaskType.fromStream(in);
  this.content = in.readBytesReference();
} else {
  // change the order as this is what write is doing in the current version
  this.taskType = TaskType.fromStream(in);
  this.content = in.readBytesReference();
}
```

Both sides of the condition are the same so I took out the transport version check, that suprised me as things are rarely this simple.  As for serverless, what was once broken should start working.